src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SortEntryConverter.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage.Search
+{
+    public class SortEntryConverter : JsonConverter
+    {
+        private static readonly ConcurrentQueue<SortValueJObjectGenerator> _cachedGenerators = new();
+
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(SortValue);
```

Comment:
Can we use this converter for the dictionary instead of each individual sort value (so newtonsoft doesn't call this method repeatedly and we don't have to rent an return a cached generator for each sortvalue)? Maybe put an attribute on the property?

Reply:
This can potentially be avoided by setting the itemtypeconverter directly (`[JsonProperty("sort", ItemConverterType = typeof(SearchIndexEntryConverter))]`)